mica/stats/guide_stats.py

```diff
@@ -0,0 +1,570 @@
+from __future__ import division
+import os
+import time
+import sys
+import argparse
+import numpy as np
+import tables
+from scipy.stats import scoreatpercentile
+from astropy.table import Table
+import logging
+import warnings
+
+from Chandra.Time import DateTime
+import agasc
+from kadi import events
+from Ska.engarchive import fetch, fetch_sci
+import mica.archive.obspar
+from mica.starcheck.starcheck import get_starcheck_catalog_at_date
+import Ska.quatutil
+import Ska.astro
+from mica.quaternion import Quat
+from chandra_aca import dark_model
+
+# Ignore known numexpr.necompiler and table.conditions warning
+warnings.filterwarnings(
+    'ignore',
+    message="using `oa_ndim == 0` when `op_axes` is NULL is deprecated.*",
+    category=DeprecationWarning)
+
+
+logger = logging.getLogger('star_stats')
+logger.setLevel(logging.INFO)
+if not len(logger.handlers):
+    logger.addHandler(logging.StreamHandler())
+
+STAT_VERSION = 0.1
+
+GUIDE_COLS = {
+    'obs': [
+        ('obsid', 'int'),
+        ('obi', 'int'),
+        ('kalman_tstart', 'float'),
+        ('kalman_datestart', 'S21'),
+        ('kalman_datestop', 'S21'),
+        ('revision', 'S15')],
+    'cat': [
+        ('slot', 'int'),
+        ('idx', 'int'),
+        ('type', 'S5'),
+        ('yang', 'float'),
+        ('zang', 'float'),
+        ('sz', 'S4'),
+        ('mag', 'float')],
+    'stat': [
+        ('n_samples', 'int'),
+        ('n_track', 'int'),
+        ('f_track', 'float'),
+        ('f_racq', 'float'),
+        ('f_srch', 'float'),
+        ('f_none', 'float'),
+        ('n_kalman', 'int'),
+        ('no_track', 'float'),
+        ('f_within_0.3', 'float'),
+        ('f_within_1', 'float'),
+        ('f_within_3', 'float'),
+        ('f_within_5', 'float'),
+        ('f_outside_5', 'float'),
+        ('f_obc_bad', 'float'),
+        ('f_common_col', 'float'),
+        ('f_quad_bound', 'float'),
+        ('f_sat_pix', 'float'),
+        ('f_def_pix', 'float'),
+        ('f_ion_rad', 'float'),
+        ('f_mult_star', 'float'),
+        ('aoacmag_min', 'float'),
+        ('aoacmag_mean', 'float'),
+        ('aoacmag_max', 'float'),
+        ('aoacmag_std', 'float'),
+        ('aoacyan_mean', 'float'),
+        ('aoaczan_mean', 'float'),
+        ('dy_min', 'float'),
+        ('dy_mean', 'float'),
+        ('dy_std', 'float'),
+        ('dy_max', 'float'),
+        ('dz_min', 'float'),
+        ('dz_mean', 'float'),
+        ('dz_std', 'float'),
+        ('dz_max', 'float'),
+        ('dr_min', 'float'),
+        ('dr_mean', 'float'),
+        ('dr_std', 'float'),
+        ('dr_5th', 'float'),
+        ('dr_95th', 'float'),
+        ('dr_max', 'float'),
+        ('n_track_interv', 'int'),
```

Comment:
I went with all "track" instead of "trak" but can obviously swap any of these back.